pombola/south_africa/views/download.py

```diff
@@ -0,0 +1,105 @@
+import os
+
+from django.db.models import Prefetch, Q
+from django.http import StreamingHttpResponse
+from django.views.generic import TemplateView
+
+import xlsx_streaming
+from pombola.core.models import Contact, Organisation, Person, Position
+
+MP_DOWNLOAD_TEMPLATE_SHEET = os.path.join(
+    os.path.dirname(os.path.realpath(__file__)), "mp-download-template.xlsx"
+)
+
+
+def generate_sheet_name(house=None):
+    """
+    Generate the sheet name of the file that will be downloaded using the house.
+    """
+    if house and house != "all":
+        return house + "-members"
+    return "members-of-parliament"
+
+
+def get_organisation_query_filter_for_house(house="all"):
+    """
+    Get the query to filter the organisations by given the house.
+    """
+    ncop_query = Q(organisation__kind__slug="provincial-legislature")
```

Comment:
wouldn't this match the western cape legislature as well?

Reply:
I just copied the [RepLocator](https://github.com/OpenUpSA/pombola/blob/master/pombola/south_africa/views/geolocalization.py#L207-L212)'s way of finding MPL positions, but I see now that [pa.org.za/organisation/ncop/people/ ](https://pa.org.za/organisation/ncop/people/) tries to find an organisation with the slug `ncop` instead.

Reply:
Changed it to get the organisation from the slug (for example, 'ncop' or 'national-assembly') now.